docs/adr/012_decouple_from_edge.md

```diff
@@ -0,0 +1,148 @@
+
+As an example, let's assume the scenario where we want to send a cloud-to-device message to a specific class-c leaf device. In this scenario we need to target the exact LNS which is managing the connection with the LoRa Basics Station in reach of the class-c device we want to target.
+
+It would be counterproductive to invoke this method on all instances we are managing and just filter it out later (depending on the DeviceID/Hostname).
+
+When scaling out, depending on the platform where the replicas of the LNS are hosted, you might have different ways to target exactly one replica.
+
+On Kubernetes, for instance, this might be achieved by using a service mesh, but this requires additional configuration
+
+#### Handle events in a Pub/Sub fashion
+
+Another alternative might be to publish the invocation events on a topic, to which all the LNS instances are subscribing and filtering based on their "hostname".
+
+This "polling" mechanism, compared to the "push" mechanism described before, might require some additional setup to "manually" invoke the methods, even though this is not the suggested way of invoking such methods.
+
+The solution might be implemented using an additional service like Azure Service Bus, or by reusing the [Azure Cache for Redis](https://docs.microsoft.com/en-us/azure/azure-cache-for-redis/cache-overview) instance that it's already being deployed in the solution as it allows the usage of [Redis Pub/Sub functionality](https://redis.io/docs/manual/pubsub/)
+
+The proposal is to **make use of Redis Pub/Sub functionality to handle remote invocation of methods in LNS**
+In addition, for high scale scenarios it could be useful to **implement a new function for clearing cache of all the LNS subscribed to the topic**
+
+## Sample "cloud" deployment scenario
+
+![Sample AKS based deployment scenario](../images/deployment-scenario-aks.png)
+
+The picture above highlights a possible scenario being enabled by decoupling LNS from IoT Edge.
+
+This scenario brings in the possibility of using one (or multiple) Azure Kubernetes Services cluster for running one (or multiple) instances of LoRaWAN Network Server.
+
+While keeping in mind that a LoRa Basics Station can be connected only to a single LNS instance at the same time, with such scenario it could be anyways possible to increase high availability or scalability.
+
+Ideally, the LoRa Basics Station could point to a ["vNext" discovery service](./009_discovery.md) that will be able to balance the load across multiple Azure Kubernetes Service clusters, or in an "active-passive" cluster configuration, only point to the "active" one.
+
+In addition, the load of the potentially multiple LNS instances in the same single cluster could be balanced by routing based on the number of "stations" already connected to the server instances (by leveraging the already existing metric being exported in Prometheus format)
+
+## Summary of the required and proposed changes
+
```

Comment:
We have a lot of proposals. Let's turn them into decisions and, with pros and cons and elaborate why we make them. Also, add the alternative to have a fixed deployment model pls and it's pros and cons.